internal/utils/utils_test.go

```diff
@@ -32,3 +32,65 @@ func TestPercentile(t *testing.T) {
 		}
 	}
 }
+
+func TestCentsToDollarStringHumanized(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		input    int
+		expected string
+	}{
+		{"1", 1, "0.01"},
+		{"10", 10, "0.10"},
+		{"100", 100, "1.00"},
+		{"1000", 1000, "10.00"},
+		{"10000", 10000, "100.00"},
+		{"100000", 100000, "1,000.00"},
+		{"1000000", 1000000, "10,000.00"},
+		{"90", 90, "0.90"},
+		{"115", 115, "1.15"},
+		{"107", 107, "1.07"},
+		{"14.5", 145, "14.5"},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			result := CentsToDollarStringHumanized(test.input)
+			if result != test.expected {
+				t.Errorf("CentsToDollarStringHumanized(%v) = %s; expected %s", test.input, result, test.expected)
+			}
+		})
+	}
+
+}
+
+func TestTrimTrailingDigits(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		inputString string
+		inputDigits int
+		expected    string
+	}{
+		{"1.00", 2, "1.00"},
```

Comment:
I think the one case we're missing is when the input is effectively an integer (like "1"), and the desired output has 2 decimal places (should return 1.00)

testing locally I see any value that contains a decimal point is working correctly (consistently two decimal places) but any whole number isn't getting `.00` appended to the end

Reply:
nm, looks like the code works correctly as is, just don't have a test case for the whole number case. merging as-is, I can add a line for the last test case. thanks for the PR!

Reply:
Ha of all the cases to have missed out, as you've said I think it should work as is, just the test is missing...